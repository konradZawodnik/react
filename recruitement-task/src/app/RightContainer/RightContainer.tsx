"use client";
import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from "@/components/ui/accordion";
import { Button } from "@/components/ui/button";
import { Label } from "@/components/ui/label";

export default function RightContainer() {
  return (
    <Accordion className="bg-white border-0 h-full max-h-[550px] max-w-[190px] mt-8 ml-4 mr-4 w-full" type="single" collapsible>
      <AccordionItem className="ml-4 mr-4 text-[#003e94]" value="firstItem">
        <AccordionTrigger>Przejdź do</AccordionTrigger>
      </AccordionItem>
      <div className="bg-[#d7d8db] border w-30"></div>
      <AccordionItem className="border-0 ml-4 mr-4" value="secondItem">
        <AccordionTrigger>Wizyta</AccordionTrigger>
        <AccordionContent>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Numer zgłoszenia
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Rodzaj wizyty
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Specjalizacja
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Data Wizyty
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Temat
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              <span>Dodatkowe</span>
              <span>informacje</span>
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Język wizyty
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Kod rabatowy
            </Label>
          </Button>
        </AccordionContent>
      </AccordionItem>
      <AccordionItem className="border-0 ml-4 mr-4" value="thirdItem">
        <AccordionTrigger>Pacjent</AccordionTrigger>
        <AccordionContent>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Kraj
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Wiek pacjenta
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Dane pacjenta
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Objawy
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Dokument
            </Label>
          </Button>
          <Button className="bg-white border-0 flex flex-row hover:bg-white justify-start pt-4 rounded-none w-full">
            <Label className="cursor-pointer font-[500] text-[14px] text-[#242628]">
              Dane adresowe
            </Label>
          </Button>
        </AccordionContent>
      </AccordionItem>
    </Accordion>
  );
}
