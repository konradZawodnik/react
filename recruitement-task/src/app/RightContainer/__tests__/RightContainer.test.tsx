import { render } from "@testing-library/react";
import RightContainer from "../RightContainer";

it("renders rightContainer unchanged", () => {
  const { container } = render(<RightContainer />);
  expect(container).toMatchSnapshot();
});
