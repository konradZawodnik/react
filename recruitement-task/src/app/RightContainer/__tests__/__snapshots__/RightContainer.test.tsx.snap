// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rightContainer unchanged 1`] = `
<div>
  <div
    class="bg-white border-0 h-full max-h-[550px] max-w-[190px] mt-8 ml-4 mr-4 w-full"
    data-orientation="vertical"
  >
    <div
      class="border-b ml-4 mr-4 text-[#003e94]"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline text-left [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          Przejd≈∫ do
          <svg
            class="lucide lucide-chevron-down h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </h3>
    </div>
    <div
      class="bg-[#d7d8db] border w-30"
    />
    <div
      class="border-0 ml-4 mr-4"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline text-left [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          Wizyta
          <svg
            class="lucide lucide-chevron-down h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="border-0 ml-4 mr-4"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline text-left [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          Pacjent
          <svg
            class="lucide lucide-chevron-down h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r4:"
        class="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r5:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
